outlines/outline.py

```diff
@@ -0,0 +1,61 @@
+import json
+from dataclasses import dataclass
+
+from pydantic import BaseModel
+
+from outlines import generate
+
+
+@dataclass
+class Outline:
+    """
+    Outline is a class that creates a callable object to generate responses
+    based on a given model, a prompt template (a function that returns a `str`) and an expected output type.
+
+    Parameters
+    ----------
+    model : object
+        The model to be used for generating responses.
+    template : function
+        A function that takes arguments and returns a prompt string.
+    output_type : type
+        The expected output type of the generated response.
+
+    Examples
+    --------
+    from pydantic import BaseModel
+    from outlines import models, Outline
+
+    class OutputModel(BaseModel):
+        result: int
+
+    model = models.transformers("gpt2")
+
+    def template(a: int) -> str:
+        return f"What is 2 times {a}?"
+
+    fn = Outline(model, template, OutputModel)
+
+    result = fn(3)
+    print(result)  # Expected output: OutputModel(result=6)
+    """
+
+    def __init__(self, model, template, output_type):
+        if not (isinstance(output_type, str) or issubclass(output_type, BaseModel)):
```

Comment:
This seems like an oversimplified check to confirm str to be a json schema string: `isinstance(output_type, str)`. Even though v1 will handle this much more gracefully, perhaps, even for a temp solution it worth to verify str being an actual json schema?

Reply:
It's surprisingly difficult to find a tool that validates a Json Schema. Would be very happy if you found one!